lib/nonblock.c

```diff
@@ -31,6 +31,9 @@
 #include <fcntl.h>
 #endif
 
+#ifdef HAVE_SYS_FILIO_H
```

Comment:
I wonder if we should invent a `__CURL_MACOS_CLASSIC__` guard in `lib/config-mac.h` and use that as well here to keep the header from not being included when not needed?

Reply:
Is your concern that including sys/filio.h when present would be a problem for some OS other than classic Mac OS?

Reply:
I was modeling my change on tests/server/tftpd.c where it uses the same condition:

```c
#ifdef HAVE_SYS_FILIO_H
/* FIONREAD on Solaris 7 */
#include <sys/filio.h>
#endif
```

But let me double-check if this is needed. Even modern macOS 10.15.7 defines `FIONBIO` in <sys/filio.h> so I was surprised curl had gotten away with not including that header here, but I see now that nonblock.c also includes <sys/ioctl.h> if present which does (on both modern macOS and in GUSI) include <sys/filio.h> already.

Reply:
Aha, even better!

Reply:
The error without my change is:

```sh
#----------------------------------------------------------
### MWC68K Compiler Error:
#	return ioctl(sockfd, FIONBIO, &flags);
#	                     ^^^^^^^
# undefined identifier 'FIONBIO'
#----------------------------------------------------------
File "nonblock.c"; Line 64
#----------------------------------------------------------
# errors caused tool to abort
make[3]: *** [nonblock.68k.o] Error 1
```

When I first saw that, I grep'd the GUSI headers to see where `FIONBIO` was defined, found it in <sys/filio.h>, found an existing include of <sys/filio.h> in nonblock.c that only took effect on NetWare (which was removed in 2d9fee450427bf20284278022f829f7cf3c05d21), and I figured I should change it to include it more broadly.

However it appears to be sufficient solely to `#define HAVE_SYS_IOCTL_H 1` in config-mac.h. Shall I change this PR to just do that? Or open a new PR to do that?

Reply:
You can update this PR, that's fine.